scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/KryoCoder.scala

```diff
@@ -27,3 +29,21 @@ object KryoCoder {
   implicit def castType[T](kryoCoder: KryoCoder): AtomicCoder[T] =
     kryoCoder.asInstanceOf[AtomicCoder[T]]
 }
+
+case class OrderedSerializationCoder[T](ord: Ordering[T])(implicit kryoCoder: KryoCoder) extends AtomicCoder[T] {
+  override def encode(value: T, outStream: OutputStream): Unit = {
+    ord match {
+      case ordSer: OrderedSerialization[T] @unchecked =>
```

Comment:
this is going to pattern match on each value. I don't think we want this. I think we should do:

```scala
object OrderedSerializationCoder {
  def apply[T](ord: Ordering[T])(implicit kryoCoder: KryoCoder): AtomicCoder[T] =
   ord match {
     case ordSer: OrderedSerialization[T] @unchecked => // make an orderedserializationcoder that ALWAYS uses ordSer
    case _ => kryoCoder
  }
}
```